pkg/docker/docker.go

```diff
@@ -310,14 +318,20 @@ func NewEngineAPIClient(config *api.DockerConfig) (*dockerapi.Client, error) {
 			},
 		}
 	}
-	return dockerapi.NewClient(config.Endpoint, os.Getenv("DOCKER_API_VERSION"), httpClient, nil)
+
+	return dockerapi.NewClientWithOpts(
```

Comment:
It took me a long time to fully understand how the CLI options interact - I'm recommend we add a comment for future maintainers.

```suggestion
        // Create a new docker client with the provided host endpoint and HTTP Client.
        // By default, this client will negotiate the Docker API version to use with the backend engine on the first request.
        // The API version to use can be fixed by setting the DOCKER_API_VERSION environment variable.
        // See https://pkg.go.dev/github.com/docker/docker/client#WithAPIVersionNegotiation and
        // https://pkg.go.dev/github.com/docker/docker/client#WithVersionFromEnv for more information
	return dockerapi.NewClientWithOpts(
```